privacyidea/api/lib/utils.py

```diff
@@ -238,21 +238,41 @@ def getLowerParams(param):
     return ret
 
 
-def get_all_params(param, body):
+def get_all_params(request):
     """
-    Combine parameters from GET and POST requests
+    Retrieve all parameters from a request, no matter if these are GET or POST requests
+    or parameters are contained as viewargs like the serial in DELETE /token/<serial>
+
+    :param request: The flask request object
     """
+    param = request.values
+    body = request.data
     return_param = {}
-    for key in param.keys():
-        return_param[key] = unquote(param[key])
-
-    # In case of serialized JSON data in the body, add these to the values.
-    try:
-        json_data = json.loads(to_unicode(body))
-        for k, v in json_data.items():
-            return_param[k] = v
-    except Exception as exx:
-        log.debug("Can not get param: {0!s}".format(exx))
+    if param:
+        log.debug(u"Update params in request {0!s} {1!s} with values.".format(request.method,
+                                                                              request.base_url))
+        # Add the unquoted HTML and form parameters
+        return_param = {key: unquote(value) for (key, value) in param.items()}
+
+    if request.json:
+        log.debug(u"Update params in request {0!s} {1!s} with JSON data.".format(request.method,
+                                                                                 request.base_url))
+        # Add the original JSON data
+        return_param.update(request.json)
+    elif body:
+        # In case of serialized JSON data in the body, add these to the values.
+        try:
+            json_data = json.loads(to_unicode(body))
+            for k, v in json_data.items():
+                return_param[k] = v
+        except Exception as exx:
+            log.debug("Can not get param: {0!s}".format(exx))
+
+    if request.view_args:
+        log.debug(u"Update params in request {0!s} {1!s} with view_args.".format(request.method,
+                                                                                 request.base_url))
+        # We add the unquoted view_args
+        return_param.update({key: unquote(value) for (key, value) in request.view_args.items()})
```

Comment:
We should also bear in mind that some parameters/args/bodies can be set by the user and thus will be available in the param dictionary. Should we check for possible dangerous parameters?

Reply:
I do not see, how this is different from the way we handled everything before?

Reply:
exactly...